waspc/cli/src/Wasp/Cli/Command/Db/Start.hs

```diff
@@ -0,0 +1,129 @@
+module Wasp.Cli.Command.Db.Start
+  ( start,
+    getDbSystem,
+    waspDevDbDockerVolumePrefix,
+  )
+where
+
+import Control.Monad (when)
+import qualified Control.Monad.Except as E
+import Control.Monad.IO.Class (liftIO)
+import Data.Maybe (fromMaybe)
+import StrongPath (Abs, Dir, File', Path', Rel, fromRelFile)
+import System.Process (callCommand)
+import Text.Printf (printf)
+import qualified Wasp.AppSpec as AS
+import qualified Wasp.AppSpec.App as AS.App
+import qualified Wasp.AppSpec.App.Db as AS.App.Db
+import qualified Wasp.AppSpec.Valid as ASV
+import Wasp.Cli.Command (Command, CommandError (CommandError))
+import Wasp.Cli.Command.Common (findWaspProjectRootDirFromCwd, throwIfExeIsNotAvailable)
+import Wasp.Cli.Command.Compile (analyze)
+import Wasp.Cli.Command.Message (cliSendMessageC)
+import Wasp.Cli.Common (WaspProjectDir)
+import qualified Wasp.Message as Msg
+import Wasp.Project.Db (databaseUrlEnvVarName)
+import Wasp.Project.Db.Dev (makeDevDbUniqueId)
+import qualified Wasp.Project.Db.Dev.Postgres as Dev.Postgres
+import Wasp.Project.Env (dotEnvServer)
+
+-- | Starts a "managed" dev database, where "managed" means that
+-- Wasp creates it and connects the Wasp app with it.
+-- Wasp is smart while doing this so it checks which database is specified
+-- in Wasp configuration and spins up a database of appropriate type.
+start :: Command ()
+start = do
```

Comment:
Uff this is weird! I would expect in this situation that `wasp start db` fails due to not being able to obtain the port 5432.
I think the solution we want here is for `wasp start db` to properly fail.
In the future we can do dynamic ports, but for now this would be the best.

The main thing to figure out is why it doesn't fail! Maybe let's go on a call together and you can show me what this looks like on your machine.

Reply:
So what Shayne said actually happens!
So we don't know, maybe it is a docker + mac thing?
I can test on LInux the same thing.
One solution could be to use non-standard port, like 5433 or something..

Reply:
I learned this is due to Docker havig its own network, and ports on host machine are forwarded to it, so native processes "steal" the stuff that comes to host machine ports and it doesn't get forwarded, if there is a native process! Interesting. I added a function to Wasp that checks if port 5432 is occupied and throws an error if it, so now for this situation `wasp start db` will throw an error! Yay!